src/main/java/ir/sahab/kafkaconsumer/SmartCommitKafkaConsumer.java

```diff
@@ -311,7 +312,7 @@ public void start() throws InterruptedException, IOException {
         ExecutorService executor = Executors.newSingleThreadExecutor();
         try {
             executor.submit(() -> {
-                kafkaConsumer.poll(0);
+                kafkaConsumer.poll(Duration.ZERO);
```

Comment:
You do not need to run it in executor! We did it because of a Kafka bug and Kafka says poll(Duration) has resolved that bug! It is even said in Javadoc of poll(Duration)

Reply:
Done